src/grid/mod.rs

```diff
@@ -300,11 +300,11 @@ impl<T: Copy + Clone> Grid<T> {
         *(self.num_lines() - line - 1)
     }
 
-    pub fn offset_to_line(&self, offset: usize) -> Option<Line> {
+    fn offset_to_line(&self, offset: usize) -> ViewportPosition {
         if offset < *self.num_lines() {
-            Some(self.lines - offset - 1)
+            ViewportPosition::Visible(self.lines - offset - 1)
         } else {
-            None
+            ViewportPosition::Above
```

Comment:
Would it make sense to remove the `offset_to_line` function altogether and put all three cases in `buffer_line_to_visible`? I find myself expecting to see three cases in one place from the definition of `ViewportPosition`.

Reply:
Originally this function had a `pub` before it for external use, which I guess was the reason it hasn't been merged with the other function yet. But since it's not used anywhere else and I've just removed the `pub` declaration, merging these two together seems like a good idea.

Thanks for the feedback.